primitives/transaction/src/historic_transaction.rs

```diff
@@ -13,6 +18,84 @@ use crate::{
     Transaction as BlockchainTransaction,
 };
 
+/// The raw transaction hash is a type wrapper.
+/// This corresponds to the hash of the transaction without the execution result.
+/// This hash is the external facing hash.
+#[derive(Clone, Debug, Eq, PartialEq, Hash)]
+pub struct RawTransactionHash {
+    hash: Blake2bHash,
+}
+
+impl From<Blake2bHash> for RawTransactionHash {
+    fn from(hash: Blake2bHash) -> Self {
+        RawTransactionHash { hash }
+    }
+}
+impl From<RawTransactionHash> for Blake2bHash {
+    fn from(hash: RawTransactionHash) -> Self {
+        hash.hash
+    }
+}
+impl Deref for RawTransactionHash {
+    type Target = Blake2bHash;
+
+    fn deref(&self) -> &Self::Target {
+        &self.hash
+    }
+}
+impl AsDatabaseBytes for RawTransactionHash {
+    fn as_database_bytes(&self) -> Cow<[u8]> {
+        Cow::Owned(self.hash.0.to_vec())
+    }
+}
+impl FromDatabaseValue for RawTransactionHash {
+    fn copy_from_database(bytes: &[u8]) -> io::Result<Self>
+    where
+        Self: Sized,
+    {
+        Ok(RawTransactionHash {
+            hash: bytes[4..].into(),
+        })
+    }
+}
+
+/// The executed transaction hash is a type wrapper.
+/// This corresponds to the hash of the transaction with the execution result.
+/// This hash is used in the history store so that we have a record of the transaction
+/// and its execution result. This is an internal to it.
+#[derive(Clone, Debug)]
+struct ExecutedTransactionHash {
+    hash: Blake2bHash,
+}
```

Comment:
Why not using a `tuple struct`?
```suggestion
struct ExecutedTransactionHash (Blake2bHash)
```